src/common/ceph_time.h

```diff
@@ -259,6 +259,14 @@ namespace ceph {
 	return time_point(seconds(ts.tv_sec) + nanoseconds(ts.tv_nsec));
       }
     };
+
+    template<typename Clock, typename Duration>
+    signedspan difference(
+      typename std::chrono::time_point<Clock, Duration> minuend,
+      typename std::chrono::time_point<Clock, Duration> subtrahend) {
+      return signedspan(minuend.time_since_epoch().count() -
```

Comment:
nit, this statement actually casts an unsigned variable to a signed one. and the width of `Duration::rep` "happen" to be identical to `signedspan`, that's why the tests pass. can we put this way?
```c++
      using rep_type = typename std::make_signed<typename Duration::rep>::type;
      return signedspan(rep_type(minuend.time_since_epoch().count()) -
                        rep_type(subtrahend.time_since_epoch().count()));
```
i think it would be, well, safer.

or we can have a
```c++
     static_assert(sizeof(typename Duration::rep) ==
                    sizeof(typename signedspan::rep), "bad duration type");
```
to make sure that the cast always works. 

what do you think?